### PR TITLE
State machine

### PR DESCRIPTION
**Pull request by benjamin-kelada**
This is the statemachine for the use of the whole arduino-uno system

### CODE REVIEW

File: statemachine.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        this->SM_state = SM_IDLE_STATE;
+        Serial.println("Transitioned from Tapping to Idle");
+      }
+      break;
+    case SM_PAUSED_STATE:
+      if (SM_event == SM_ONE) {
+        this->SM_state = SM_TAPPING_STATE;
+        Serial.println("Transitioned from Paused to Tapping");
+      } 
+      }
```
Comment (brian-vdb) [inline statemachine.cpp:38]: Vanaf dit punt moet de code 1 naar links geshift worden

**Review by brian-vdb — CHANGES_REQUESTED** [review]
Verander wat de comment zegt

File: statemachine.cpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+// Function to handle SM_state transitions
+void StateMachine::handleInputEvent(int SM_event) {
+  switch (SM_state) {
+    case SM_IDLE_STATE:
+      if (SM_event == SM_ONE) {
+        this->SM_state = SM_TAPPING_STATE;
+        Serial.println("Transitioned from Idle to Tapping");
+      }
+      // Stay in SM_IDLE_STATE if SM_ZERO is received
```
Comment (brian-vdb) [inline statemachine.cpp:22]: Either comment every state or remove this one

File: statemachine.cpp [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        Serial.println("Transitioned from Tapping to Idle");
+      }
+      break;
+    case SM_PAUSED_STATE:
+      if (SM_event == SM_ONE) {
+        this->SM_state = SM_TAPPING_STATE;
+        Serial.println("Transitioned from Paused to Tapping");
+      } 
+      break;
+
```
Comment (brian-vdb) [inline statemachine.cpp:39]: Code consistency means that you should remove this extra enter or add one to the other comments

**Review by brian-vdb — CHANGES_REQUESTED** [review]

File: statemachine.cpp [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+StateMachine::StateMachine() {
+  this->SM_state = SM_IDLE_STATE;
+}
+
+int StateMachine::getState() const {
+    return this->SM_state; // Return the current state
+}
+
+// Function to handle SM_state transitions
+void StateMachine::handleInputEvent(int SM_event) {
```
Comment (brian-vdb) [inline statemachine.cpp:15]: Geen comments

File: StateMachine.h [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+// Define states as constants
+#define SM_IDLE_STATE 0
+#define SM_TAPPING_STATE 1
+#define SM_PAUSED_STATE 2
+
+// Define events as constants
+#define SM_ZERO 0
+#define SM_ONE 1
+
+class StateMachine {
```
Comment (brian-vdb) [inline StateMachine.h:17]: Geen function descriptors

**Review by brian-vdb — APPROVED** [review]

**Review by Diego-Brandjes — APPROVED** [review]

**Pull request merged**